lightfm/datasets/movielens.py

```
@@ -4,11 +4,224 @@
+    
+    def getnnz(self, axis=None):
+        """Get number of non-zero elements along given axis"""
+        import numpy as np
+        
+        if axis is None:
+            # Total number of non-zero elements
+            return len(self.data)
+        elif axis == 0:
+            # Number of non-zeros per column
+            col_nnz = {}
+            for row, col in zip(self.row, self.col):
```

Ruff (0.11.9)

Loop control variable `row` not used within loop body

(B007)